### DISCUSSION TITLE
Framerate and stuttering issues becoming more important as I progress

### DISCUSSION THREAD

**Discussion by victorleblais** [voxeltycoon/issues#566] [General]
**Describe the bug**
The longer I play and do stuff (build trains and trucks and all), the more the performance seems to degrade. I'm starting to have issues when running at triple speed, even double speed starts to stutter a little bit.

I'm really worried about performance and how it'll end up if I keep playing.

**Steps to reproduce**
Have a large game with many vehicles and stations.

**Additional Resources**
Here's a video recording, you can see the difference in camera responsivity and framerate smoothness when going from normal speed to triple speed. Please note that not long ago, when my game was less developed, triple was as smooth as normal speed.
https://streamable.com/iey6n

Here's my current save game: https://we.tl/t-W61c8S0Cm9
(notify me if link is dead)

Here's a screenshot of resources usage (there's plenty of headroom !):
![image](https://user-images.githubusercontent.com/4365075/60389736-87dba900-9ac7-11e9-8ed3-3c4943c508f1.png)
(temps are high since weather reached around 42°C the last few days)

**Build version**
0.71 2019-06-26 ITCH PRE-ALPHA

**Platform**
Windows 10 Pro N (regular Windows but without Windows Media Player)
Intel i7-8700k, 6 cores, 12 threads at 4.4Ghz
32GB of RAM, DDR4, 3466Mhz
GTX 1080Ti

**Reproducibility**
Restarting the game changes nothing, so I'd say that it's inevitable if you play the game long enough

**Comment by katepolak** [comment]
This is unfortunately known issue and is one of the main reasons the game is in pre-alpha not early access.

**Comment by katepolak** [comment]
Some improvements for sped up gameplay implemented in 0.72.

**Comment by victorleblais** [comment]
I can confirm that it runs smoother in 0.72 (and now 0.73) improved the situation quite a bit; but it's still not perfectly smooth, although I haven't looked again at the cpu/gpu/ram usage.

**Comment by victorleblais** [comment]
I can confirm that even on newer patches (currently 0.76), the game still does not use the most of your computer's resources:

You can have the game use less than 50% CPU, less than 50% GPU, less than 50% RAM, less than 50% storage bandwidth (read or write) and the game still won't reach high framerates...

If you have a monitor with a refresh rate of 120Hz or more, this is especially obvious. But I've seen people on Discord reporting that they barely reach 30-40fps despite having plenty of pc resources available.

**Comment by parakanastika** [comment]
I can confirm the fact that the game is not using the available resources. (v 0.76)
I have AMD 2700x 16GB/3200 , GPU AMD RX 480 /8GB . The processor is at 25%, and GPU at 25% usage. In this time the game is running at ~30 FPS on a 100 vehicle map, there is no V-Sync available and there is tearing also.

**Comment by parakanastika** [comment]
Started new game, just for test, nothing build. 160FPS / GPU 90%, CPU 45%, 2560x1440@144Hz - for max zoom out. 240 FPS for max zoom in.

**Comment by katepolak** [comment]
I see that behavior in a lot of games, but I don't really see any common factors that could cause it. Different engines, different devs, different everything.

On one hand I want the game to run as fast as possible but on the other hand I don't want it to use up all my resources as I want to have a video running on the second screen or something like that. That said, I often see the issue that I get huge FPS drops without using all of the CPU. Like the CPU getting used only like 60% overall but won't get higher. This is quite likely a CPU issue though because it's quite crap.

**Comment by miohh** [comment]
I get more and more jank as the game goes on too in 0.86.0. Less than a single core of cpu usage, 25% gpu usage, 1/2 ram used. It's most noticeable scrolling or completing any click commands. The pre "save game pause" pause get's longer too, causing more wayward clicks to be registered post pause.

Understandable that it's still early but this type of game attracts the kind of long term player so seem's like a fairly important inefficiency to address.

**Comment by tycode** [comment]
I have 40 regions, ~500 trucks and ~150 trains, and I'm not planning on stopping expanding any time soon :)

The game runs absolutely fine when it's paused, but when it's running, it runs at about 2fps. GPU usage goes way down, and CPU is bottlenecked on a single thread.

Now I'm not particularly bothered about the game state updating at 60fps. In fact, I'm perfectly happy with the game state updating at 2fps. But what I'd like to see, is the simulation logic moved to a separate thread, so it can take as long as it likes while the UI and camera movement runs at a full 60fps just like it does when the game is paused.

When in "take a ride" mode, linear interpolation of the vehicle position to the most recently calculated from the one before it would be fine to make it look smooth.

**Reply by miohh** [reply]
I like the UI rendering a view of the current, separate game state idea. Let the simulation chug along doing it's thing in it's own time. I'm guessing difficult to do and adds some overhead, I guess game time would dilate or something.

**Comment by pilotflying777** [comment]
I can confirm that issue the same way as tycode or miohh described it. This lags and frame rate drops are highly influencing fun while gaming and need to be addressed, as the game makes so much more fun when you are progressing. Therefore please improve the performance as best as possible and enable long term attraction.

**Reply by pilotflying777** [reply]
Addition: I have version 0.86 on Windows 10 in place.

**Comment by jmax1978** [comment]
I have the same problem , currently with 220 trains , I do have a solid RIG with 2 way SLI RTX 3070 64 GB RAM , my cpu load 49% load of the GPU is 24% avg each  , game is lagging and I get only 2 fps barely can play. I would like to keep building but this is a turn off , doesn’t make any sense having a overkill RIG and still having performance issue from the game at late game!

**Reply by pilotflying777** [reply]
I fully agree with you. The situation is very similar in my case as I have a High End System with 2 Way SLI RTX 2080 Ti and an I9 10900K CPU together with 64 GB RAM. I would highly appreciate a performance update addressing this issues. Nevertheless it is a strange situation that none of the PC components is fully occupied but FPS numbers are so low that you are unable to play.

**Reply by jmax1978** [reply]
Indeed, I think there not many players that have reach our threshold where they start experience the same issue.  Only happens  late in the game . They only way to play it is on pause mode so you can really build stuff, and at normal speed and still is lagging, one thing that improve a bit, is if you put voxel.exe High Priority on the Detail Tab on the Task Manager.
